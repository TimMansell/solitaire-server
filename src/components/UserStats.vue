<template>
  <div>
    Games: <span data-test="stats">{{ userStats.gameNumber }}</span> (<Button
      link
      @click="viewStats"
      data-test="view-stats"
    >
      view stats </Button
    >)
    <UserStatsOverlay v-if="showStats" data-test="user-stats" />
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';
import UserStatsOverlay from '@/components/UserStatsOverlay.vue';
import Button from '@/components/Button.vue';

export default {
  name: 'Stats',
  components: {
    UserStatsOverlay,
    Button,
  },
  computed: {
    ...mapGetters(['userStats', 'showStats']),
  },
  methods: {
    ...mapActions(['toggleStats', 'setGamePaused']),
    viewStats() {
      const isPaused = {
        isPaused: true,
        isActive: false,
        showMsg: false,
      };

      this.setGamePaused(isPaused);
      this.toggleStats();
    },
  },
};
</script>
