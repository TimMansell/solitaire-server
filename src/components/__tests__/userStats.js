import { shallowMount } from '@vue/test-utils';
import UserStats from '@/components/UserStats.vue';

describe('UserStats.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(UserStats, {
      computed: {
        userStats: () => ({
          gameNumber: 1,
        }),
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should show 1 game played', () => {
    const wrapper = shallowMount(UserStats, {
      computed: {
        userStats: () => ({
          gameNumber: 1,
        }),
      },
    });

    expect(wrapper.find('[data-test="stats"]').text()).toContain('1');
  });
});
