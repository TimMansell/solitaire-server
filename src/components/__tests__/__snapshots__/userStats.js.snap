// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStats.vue matches snapshot 1`] = `
<div>
  Games: <span data-test="stats">1</span></div>
`;
